pkg/vsphere/tasks/waiter.go

```diff
@@ -65,20 +73,80 @@ func Wait(ctx context.Context, f func(context.Context) (Waiter, error)) error {
 func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
 	task, err := f(ctx)
 	if err != nil {
-		cerr := errors.Errorf("Failed to invoke operation: %s", errors.ErrorStack(err))
-		log.Errorf(cerr.Error())
-		return nil, cerr
+		terr := &TaskError{
```

Comment:
because the current way other functions rely on this function they do not expect an info object back on a failure. if we want to see the fault our two choices are to return the info even on error, or create a new error type that has the LocalizedMethodFault packed into it.